services-core/src/solution_submission.rs

```diff
@@ -161,14 +157,12 @@ impl<'a> StableXSolutionSubmitter<'a> {
 
     async fn cancel_transaction_after_deadline(
         &self,
```

Comment:
A potential future change is to move cancellation into `retry` where it fits better than here.